docs/requirements.txt

```diff
@@ -1,7 +1,258 @@
+    --hash=sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
+    # via
+    #   fake-factory
+    #   jsonschema
+    #   python-dateutil
+    #   sphinx
+    #   sphinxcontrib-httpdomain
+snowballstemmer==2.1.0 \
+    --hash=sha256:b51b447bea85f9968c13b650126a888aabd4cb4463fca868ec596826325dedc2 \
+    --hash=sha256:e997baa4f2e9139951b6f4c631bad912dfd3c792467e2f03d7239464af90e914
+    # via sphinx
+sphinx==1.8.5 \
+    --hash=sha256:9f3e17c64b34afc653d7c5ec95766e03043cc6d80b0de224f59b6b6e19d37c3c \
+    --hash=sha256:c7658aab75c920288a8cf6f09f244c6cfdae30d82d803ac1634d9f223a80ca08
+    # via
+    #   -r requirements.in
+    #   guzzle-sphinx-theme
+    #   sphinx-jsondomain
+    #   sphinxcontrib-httpdomain
+https://github.com/pieterlexis/sphinx-jsondomain/archive/refs/heads/no-type-links.zip \
+    --hash=sha256:0df65f5b93902e9e4f03935f39d91ae2a6a782116078f8af99ec1e370f999257
+    # via
+    #   -r requirements.in
+    #   sphinxcontrib-openapi
+sphinxcontrib-fulltoc==1.2.0 \
+    --hash=sha256:c845d62fc467f3135d4543e9f10e13ef91852683bd1c90fd19d07f9d36757cd9
+    # via -r requirements.in
+sphinxcontrib-httpdomain==1.8.0 \
+    --hash=sha256:2059cfabd0cca8fcc3455cc8ffad92f0915a7d3bb03bfddba078a6a0f35beec5 \
+    --hash=sha256:a3396d6350728d574f52458b400f0ac848f8b6913bd41fed95d391d3ffbbade3
+    # via sphinxcontrib-openapi
+https://github.com/Habbie/sphinxcontrib-openapi/archive/refs/heads/use-jsondomain-pdns.zip \
+    --hash=sha256:d2993cb81038368233b40f839f885d9d535af2570611b1185bfc716097671d7c
+    # via -r requirements.in
+sphinxcontrib-serializinghtml==1.1.5 \
+    --hash=sha256:352a9a00ae864471d3a7ead8d7d79f5fc0b57e8b3f95e9867eb9eb28999b92fd \
+    --hash=sha256:aa5f6de5dfdf809ef505c4895e51ef5c9eac17d0f287933eb49ec495280b6952
+    # via sphinxcontrib-websupport
+sphinxcontrib-websupport==1.2.4 \
+    --hash=sha256:4edf0223a0685a7c485ae5a156b6f529ba1ee481a1417817935b20bde1956232 \
+    --hash=sha256:6fc9287dfc823fe9aa432463edd6cea47fa9ebbf488d7f289b322ffcfca075c7
+    # via sphinx
+typing-extensions==3.10.0.2 \
+    --hash=sha256:49f75d16ff11f1cd258e1b988ccff82a3ca5570217d7ad8c5f48205dd99a677e \
+    --hash=sha256:d8226d10bc02a29bcc81df19a26e56a9647f8b0a6d4a83924139f4a8b01f17b7 \
+    --hash=sha256:f1d25edafde516b146ecd0613dabcc61409817af4766fbbcfb8d1ad4ec441a34
+    # via importlib-metadata
+urllib3==1.26.7 \
+    --hash=sha256:4987c65554f7a2dbf30c18fd48778ef124af6fab771a377103da0585e2336ece \
+    --hash=sha256:c4fdf4019605b6e5423637e01bc9fe4daef873709a7973e195ceba0a62bbc844
+    # via requests
+zipp==3.5.1 \
+    --hash=sha256:1fc9641b26f3bd81069b7738b039f2819cab6e3fc3399a953e19d92cc81eff4d \
+    --hash=sha256:8dc6c4d5a809d659067cc713f76bcf42fae8ae641db12fddfa93694a15abc96b
+    # via importlib-metadata
+
+# WARNING: The following packages were not pinned, but pip requires them to be
+# pinned when the requirements file includes hashes. Consider using the --allow-unsafe flag.
+# setuptools
```

Comment:
Is this something we need to worry about?

Reply:
It's a terribly confusing story. `--allow-unsafe` will become the default in the next version of `pip-tools` because ... that's the safe choice.

In this specific case, we never install setuptools (`python3 -mvenv` does that, from the distro version I believe, which I will check), so there indeed is nothing to worry about.

Reply:
Good, thank you :)

Reply:
For context, this is what `--allow-unsafe` does:
```
-# WARNING: The following packages were not pinned, but pip requires them to be
-# pinned when the requirements file includes hashes. Consider using the --allow-unsafe flag.
-# setuptools
+# The following packages are considered to be unsafe in a requirements file:
+setuptools==58.1.0 \
+    --hash=sha256:5de67252090e08d25f240f07d80310f778a5a46cdcf9ea9855662630ac8547b2 \
+    --hash=sha256:7324fd4b66efa05cdfc9c89174573a4410acc7848f318cc0565c7fb659dfdc81
+    # via sphinx
```

so depending on whether we use the flag, setuptools is not pulled in, or it is pinned.